osquery/events/linux/auditdnetlink.cpp

```diff
@@ -779,6 +798,11 @@ bool AuditdNetlinkParser::ParseAuditReply(
     return true;
   }
 
+  // Save the whole message for AppArmor too
+  if (isAppArmorRecord(reply)) {
+    event_record.raw_data = reply.message;
```

Comment:
The SELinux check above returns early. I am double-checking that you do not want to do the same thing for AppArmor?

Reply:
> The SELinux check above returns early. I am double-checking that you do not want to do the same thing for AppArmor?

I double checked this place and it's just because osquery doesn't parse and tokenise SELinux messages so thet's why the early return exists. As you can see that in the [schema](https://osquery.io/schema/4.4.0/#selinux_events) - SELinux message is saved as a string.

For AppArmor I'd like to parse the message and have it tokenized to be able to do joins on uid, pid etc.